app/src/main/java/com/example/taskyy/data/repositories/AuthRepositoryImpl.kt

```diff
@@ -23,14 +24,20 @@ class AuthRepositoryImpl @Inject constructor(
         }
     }
 
-    override fun registerUser(registerUserDTO: RegisterUserDTO) {
+    override fun registerUser(user: User) {
+        val mapper = UserToUserDTOMapper()
         runBlocking { launch {
             val response = try {
-                retrofit.registerUser(registerUserDTO)
+                retrofit.registerUser(mapper.mapUserToUserDTO(user = user))
             } catch (e: HttpException){
                 Log.e("TAG", "HttpException, " + e.printStackTrace())
             }
             Log.e("TAG", response.toString())
         } }
     }
+
+    override fun validatePassword(password: String): Boolean {
+        val passwordRegex = "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{9,}$"
+        return passwordRegex.toRegex().matches(password)
+    }
```

Comment:
For simple patterns like a valid password, I'm always a friend of doing this manually since it lets you quickly know what makes a valid password. Something like this: 
```
val hasLowerCase = password.any { it.isLowerCase() }
val hasUpperCase = password.any { it.isUpperCase() }
val hasDigit = password.any { it.hasDigit() }
val hasValidLength = password.length in min..max

return hasLowerCase && hasUpperCase && hasDigit && hasValidLength
```
That way, probably even my grandma could tell you what the requirements for a valid password are :D 
Regex makes sense if you can't easily do this manually, but has the cost of making code very unreadable.